@wordpress/element typings reference @types/react-dom, but it doesn't have it as a dependency

**Describe the bug**
Hi. I am developing WordPress plugins using TypeScript and after the recent deprecation of `@types/wordpress__element`, I tried to switch to the type definitions included in `@wordpress/element` - I've installed `@wordpress/element` as a dev-dependency and included the following type declaration in my project:
```ts
declare const wp: {
	components: typeof import('wordpress__components'); // That's how it was done before, for reference
	element: typeof import('@wordpress/element');
};
```

However, I got the error
```
/home/user/project/node_modules/@wordpress/element/build-types/react-platform.d.ts(1,30): error TS7016: Could not find a declaration file for module 'react-dom'. '/home/user/project/node_modules/react-dom/index.js' implicitly has an 'any' type.
  Try `npm install @types/react-dom` if it exists or add a new declaration (.d.ts) file containing `declare module 'react-dom';`
```

Upon closer inspection, I've found out that the type definitions from `@wordpress/element` reference types for `react-dom`, but it has `@types/react-dom` only as a dev-dependency. I believe this is a bug and `@types/react-dom` should be a dependency for ` @wordpress/element` so that the type definitions are complete.

- `@wordpress/element` version: `2.14.0`
 - OS: Debian testing

Thanks!

--- a/package.json
+++ b/package.json
@@ -102,7 +102,6 @@
 		"@types/npm-package-arg": "6.1.0",
 		"@types/prettier": "1.19.0",
 		"@types/qs": "6.9.1",
-		"@types/react-dom": "16.9.5",
 		"@types/requestidlecallback": "0.3.1",
 		"@types/semver": "7.2.0",
 		"@types/sprintf-js": "1.1.2",

--- a/packages/element/CHANGELOG.md
+++ b/packages/element/CHANGELOG.md
@@ -2,6 +2,11 @@
 
 ## Unreleased
 
+### Bug fix
+
+- Declare @types/react and @types/react-dom dependencies which could cause type errors when using
+  this package with TypeScript ([#25086](https://github.com/WordPress/gutenberg/pull/25086))
+
 ## 2.14.0 (2020-05-14)
 
 ### New Feature

--- a/packages/element/package.json
+++ b/packages/element/package.json
@@ -4,12 +4,7 @@
 	"description": "Element React module for WordPress.",
 	"author": "The WordPress Contributors",
 	"license": "GPL-2.0-or-later",
-	"keywords": [
-		"wordpress",
-		"gutenberg",
-		"element",
-		"react"
-	],
+	"keywords": [ "wordpress", "gutenberg", "element", "react" ],
 	"homepage": "https://github.com/WordPress/gutenberg/tree/master/packages/element/README.md",
 	"repository": {
 		"type": "git",
@@ -26,6 +21,8 @@
 	"sideEffects": false,
 	"dependencies": {
 		"@babel/runtime": "^7.9.2",
+		"@types/react": "^16.9.0",
+		"@types/react-dom": "^16.9.0",
 		"@wordpress/escape-html": "file:../escape-html",
 		"lodash": "^4.17.19",
 		"react": "^16.13.1",

